ui/pages/confirmations/components/confirm/info/shared/transaction-details/transaction-details.tsx

```diff
@@ -179,13 +179,19 @@ export const TransactionDetails = () => {
   if (currentConfirmation?.type === TransactionType.revokeDelegation) {
     return null;
   }
+  const { nestedTransactions, txParams } = currentConfirmation ?? {};
+  const { from, to } = txParams ?? {};
+
+  const isBatch =
+    Boolean(nestedTransactions?.length) &&
+    to?.toLowerCase() === from.toLowerCase();
```

Comment:
Utility fn can be used here

Reply:
Pushed the utility function [here](https://github.com/MetaMask/metamask-extension/pull/31840/commits/8183ff3032b58aaada21062681d194397e422b3e)